schema/schema.go

```diff
@@ -180,6 +190,29 @@ func addPrimaryKeyIfMissing(reqSchema jsoniter.RawMessage) (jsoniter.RawMessage,
 	return jsoniter.Marshal(schema)
 }
 
+func setPrimaryKeyForMessages(reqSchema jsoniter.RawMessage) (jsoniter.RawMessage, error) {
```

Comment:
Possible to consolidate this with `addPrimaryKeyIfMissing` method? or extract out the common functionality? Looks like the idea of both the methods is the same except for the type of the id field.